src/app/components/Timer.js

```diff
@@ -0,0 +1,28 @@
+export const convertSeconds = (sec) => {
+    let minutes = Math.floor(sec / 60);
+    minutes = minutes < 10 ? `0${minutes}` : minutes;
+    let seconds = sec % 60;
+    seconds = seconds < 10 ? `0${seconds}` : seconds;
+    return { minutes, seconds };
+};
+
+export const generateTimer = ({ timeleft, onTimerEnd }) => {
+    if (timeleft < 0) throw new Error('Time can not be negative!');
+    if (timeleft == 0) return onTimerEnd();
```

Comment:
To jest dość dziwne zachowanie... Nie tworzysz komponentu, ale wywołujesz callback `onTimerEnd`. Może najlepiej byłoby rzucić Error gdy `timeleft <= 0`

Reply:
Teraz zwraca element z czasem 0 i wywołuje callback